### PR TITLE
implement cleanup tasks as a @PreDestroy component

### PR DESCRIPTION
**Pull request by mbaechler**
I will link it to JAMES-2470 once green

### CODE REVIEW

File: server/container/guice/guice-common/src/main/java/org/apache/james/CleanupTasksPerformer.java [hunk 1]
```diff
@@ -39,6 +40,7 @@ public CleanupTasksPerformer(Set<CleanupTask> cleanupTasks) {
         this.cleanupTasks = cleanupTasks;
     }
 
+    @PreDestroy
```
Comment (chibenwa) [inline server/container/guice/guice-common/src/main/java/org/apache/james/CleanupTasksPerformer.java]: Shouldn't it be done on a per-task basis?

**Comment by trantienduchn** [comment]
```
[9476bf1b86c7aad262ce3fa4c945ac9ca655c677] [ERROR] Errors: 
[9476bf1b86c7aad262ce3fa4c945ac9ca655c677] [ERROR]   GuiceLifecycleHeathCheckTest$Unhealthy.stoppingJamesServerShouldBeUnhealthy:117 » Connect
[9476bf1b86c7aad262ce3fa4c945ac9ca655c677] [INFO] 
[9476bf1b86c7aad262ce3fa4c945ac9ca655c677] [ERROR] Tests run: 16, Failures: 0, Errors: 1, Skipped: 0
```
test this please

File: server/container/guice/guice-common/src/main/java/org/apache/james/CleanupTasksPerformer.java (hunk 1 above)
Comment (mbaechler) [reply server/container/guice/guice-common/src/main/java/org/apache/james/CleanupTasksPerformer.java]: There are only classes that are not injected anywhere so my reasoning is that they are created after their dependencies (think, Cassandra cleanup task requires Cassandra so will be created after Cassandra) and will then be called before application component without any ordering.
I think this is more than enough and we don't need to change the registration process of every cleanup tasks.
WDYT ?

**Comment by mbaechler** [comment]
```

[9476bf1b86c7aad262ce3fa4c945ac9ca655c677] [ERROR] Tests run: 2, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 6.71 s <<< FAILURE! - in org.apache.james.GuiceLifecycleHeathCheckTest
[9476bf1b86c7aad262ce3fa4c945ac9ca655c677] [ERROR] stoppingJamesServerShouldBeUnhealthy{GuiceJamesServer}  Time elapsed: 5.602 s  <<< ERROR!
[9476bf1b86c7aad262ce3fa4c945ac9ca655c677] java.net.ConnectException: Connection refused (Connection refused)
[9476bf1b86c7aad262ce3fa4c945ac9ca655c677] 	at org.apache.james.GuiceLifecycleHeathCheckTest$Unhealthy.stoppingJamesServerShouldBeUnhealthy(GuiceLifecycleHeathCheckTest.java:117)
```

I will investigate

File: server/container/guice/guice-common/src/main/java/org/apache/james/CleanupTasksPerformer.java (hunk 1 above)
Comment (chibenwa) [reply server/container/guice/guice-common/src/main/java/org/apache/james/CleanupTasksPerformer.java]: I am not sure a single point where all cleanUp task can be called safely...

Comment (mbaechler) [reply server/container/guice/guice-common/src/main/java/org/apache/james/CleanupTasksPerformer.java]: what do you mean by safely ?

**Comment by mbaechler** [comment]
test this please

**Comment by mbaechler** [comment]
```
[a5602693bc6aa2067389a5e1ab8ad08b48fcdca5] [ERROR] Errors: 
[a5602693bc6aa2067389a5e1ab8ad08b48fcdca5] [ERROR]   FilterTest.setup:86 » DomainList domain.tld already exists.
[a5602693bc6aa2067389a5e1ab8ad08b48fcdca5] [ERROR]   FilterTest.setup:86 » DomainList domain.tld already exists.
```

cleanup is probably not registered properly, I'm looking at that

**Comment by mbaechler** [comment]
``
[a265daa5bc563675b695bd3598faabe71f49edef] [INFO] Apache James :: Server :: JMAP :: RabbitMQ + Swift + Cassandra Integration testing FAILURE [04:21 min]
```

We are already paying the price of code duplication (Cassandra and Rabbit product were supposed to merge fast but as it's not the case, we need to do things twice).

**Comment by mbaechler** [comment]
```

[9fe8a59de4d9942199ae0065178dab61616945b0] [ERROR] Error occurred in starting fork, check output in log
[9fe8a59de4d9942199ae0065178dab61616945b0] [ERROR] Process Exit Code: 100
[9fe8a59de4d9942199ae0065178dab61616945b0] [ERROR] Crashed tests:
[9fe8a59de4d9942199ae0065178dab61616945b0] [ERROR] org.apache.james.jmap.memory.MemoryForwardIntegrationTest
[9fe8a59de4d9942199ae0065178dab61616945b0] [ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.fork(ForkStarter.java:671)
[9fe8a59de4d9942199ae0065178dab61616945b0] [ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.fork(ForkStarter.java:533)
[9fe8a59de4d9942199ae0065178dab61616945b0] [ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.access$600(ForkStarter.java:115)
[9fe8a59de4d9942199ae0065178dab61616945b0] [ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter$2.call(ForkStarter.java:429)
[9fe8a59de4d9942199ae0065178dab61616945b0] [ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter$2.call(ForkStarter.java:406)
[9fe8a59de4d9942199ae0065178dab61616945b0] [ERROR] 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[9fe8a59de4d9942199ae0065178dab61616945b0] [ERROR] 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[9fe8a59de4d9942199ae0065178dab61616945b0] [ERROR] 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
```
test this please

**Comment by mbaechler** [comment]
```

[9fe8a59de4d9942199ae0065178dab61616945b0] [ERROR] Errors: 
[9fe8a59de4d9942199ae0065178dab61616945b0] [ERROR]   GetMailboxesMethodTest.setup:120 » DomainList domain.tld already exists.
[9fe8a59de4d9942199ae0065178dab61616945b0] [ERROR]   RabbitMQDownloadCucumberTest » DomainList domain.tld already exists.
[9fe8a59de4d9942199ae0065178dab61616945b0] [ERROR]   RabbitMQDownloadCucumberTest » DomainList domain.tld already exists.
[9fe8a59de4d9942199ae0065178dab61616945b0] [ERROR]   RabbitMQDownloadCucumberTest » DomainList domain.tld already exists.
[9fe8a59de4d9942199ae0065178dab61616945b0] [ERROR]   RabbitMQDownloadCucumberTest » DomainList domain.tld already exists.
[9fe8a59de4d9942199ae0065178dab61616945b0] [ERROR]   RabbitMQGetMessagesMethodTest » DomainList domain.tld already exists.
[9fe8a59de4d9942199ae0065178dab61616945b0] [ERROR]   RabbitMQGetMessagesMethodTest » DomainList domain.tld already exists.
[9fe8a59de4d9942199ae0065178dab61616945b0] [ERROR]   RabbitMQGetMessagesMethodTest » DomainList domain.tld already exists.
[9fe8a59de4d9942199ae0065178dab61616945b0] [ERROR]   RabbitMQMailboxSharingTest » DomainList domain.tld already exists.
[9fe8a59de4d9942199ae0065178dab61616945b0] [ERROR]   RabbitMQMessageSharingTest » DomainList domain.tld already exists.
[9fe8a59de4d9942199ae0065178dab61616945b0] [ERROR]   RabbitMQMessageSharingTest » DomainList domain.tld already exists.
[9fe8a59de4d9942199ae0065178dab61616945b0] [ERROR]   RabbitMQMessageSharingTest » DomainList domain.tld already exists.
[9fe8a59de4d9942199ae0065178dab61616945b0] [ERROR]   RabbitMQMessageSharingTest » DomainList domain.tld already exists.
[9fe8a59de4d9942199ae0065178dab61616945b0] [ERROR]   RabbitMQMessageSharingTest » DomainList domain.tld already exists.
[9fe8a59de4d9942199ae0065178dab61616945b0] [ERROR]   RabbitMQMessageSharingTest » DomainList domain.tld already exists.
[9fe8a59de4d9942199ae0065178dab61616945b0] [ERROR]   RabbitMQMessageSharingTest » DomainList domain.tld already exists.
[9fe8a59de4d9942199ae0065178dab61616945b0] [ERROR]   RabbitMQMessageSharingTest » DomainList domain.tld already exists.
[9fe8a59de4d9942199ae0065178dab61616945b0] [ERROR]   RabbitMQMessageSharingTest » DomainList domain.tld already exists.
[9fe8a59de4d9942199ae0065178dab61616945b0] [ERROR]   RabbitMQMessageSharingTest » DomainList domain.tld already exists.
[9fe8a59de4d9942199ae0065178dab61616945b0] [ERROR]   RabbitMQMessageSharingTest » DomainList domain.tld already exists.
[9fe8a59de4d9942199ae0065178dab61616945b0] [ERROR]   RabbitMQSetMailboxesMethodCucumberTest » DomainList domain.tld already exists.
[9fe8a59de4d9942199ae0065178dab61616945b0] [ERROR]   RabbitMQSetMailboxesMethodCucumberTest » DomainList domain.tld already exists.
[9fe8a59de4d9942199ae0065178dab61616945b0] [ERROR]   RabbitMQSetMessagesMethodCucumberTest » DomainList domain.tld already exists.
[9fe8a59de4d9942199ae0065178dab61616945b0] [ERROR]   RabbitMQSetMessagesMethodCucumberTest » DomainList domain.tld already exists.
[9fe8a59de4d9942199ae0065178dab61616945b0] [ERROR]   RabbitMQUploadCucumberTest » DomainList domain.tld already exists.
```

**Comment by mbaechler** [comment]
```

[1e8e6ea84f4a7a5c063a35d19b5ed4b3e9a8c39e] [ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 64.347 s <<< FAILURE! - in org.apache.james.mpt.smtp.RabbitMQForwardSmtpTest
[1e8e6ea84f4a7a5c063a35d19b5ed4b3e9a8c39e] [ERROR] forwardingAnEmailShouldWork  Time elapsed: 62.031 s  <<< ERROR!
[1e8e6ea84f4a7a5c063a35d19b5ed4b3e9a8c39e] org.awaitility.core.ConditionTimeoutException: 
[1e8e6ea84f4a7a5c063a35d19b5ed4b3e9a8c39e] Assertion condition defined as a org.apache.james.mpt.smtp.ForwardSmtpTest 1 expectation failed.
[1e8e6ea84f4a7a5c063a35d19b5ed4b3e9a8c39e] JSON path [0].from doesn't match.
[1e8e6ea84f4a7a5c063a35d19b5ed4b3e9a8c39e] Expected: matthieu@yopmail.com
[1e8e6ea84f4a7a5c063a35d19b5ed4b3e9a8c39e]   Actual: null
[1e8e6ea84f4a7a5c063a35d19b5ed4b3e9a8c39e]  within 60 seconds.
[1e8e6ea84f4a7a5c063a35d19b5ed4b3e9a8c39e] 	at org.apache.james.mpt.smtp.ForwardSmtpTest.forwardingAnEmailShouldWork(ForwardSmtpTest.java:78)
[1e8e6ea84f4a7a5c063a35d19b5ed4b3e9a8c39e] Caused by: java.lang.AssertionError: 
[1e8e6ea84f4a7a5c063a35d19b5ed4b3e9a8c39e] 1 expectation failed.
[1e8e6ea84f4a7a5c063a35d19b5ed4b3e9a8c39e] JSON path [0].from doesn't match.
[1e8e6ea84f4a7a5c063a35d19b5ed4b3e9a8c39e] Expected: matthieu@yopmail.com
[1e8e6ea84f4a7a5c063a35d19b5ed4b3e9a8c39e]   Actual: null
[1e8e6ea84f4a7a5c063a35d19b5ed4b3e9a8c39e] 
[1e8e6ea84f4a7a5c063a35d19b5ed4b3e9a8c39e] 	at org.apache.james.mpt.smtp.ForwardSmtpTest.lambda$null$0(ForwardSmtpTest.java:80)
[1e8e6ea84f4a7a5c063a35d19b5ed4b3e9a8c39e] 	at org.apache.james.mpt.smtp.ForwardSmtpTest.lambda$forwardingAnEmailShouldWork$1(ForwardSmtpTest.java:79)
```

test this please

**Comment by mbaechler** [comment]
test this please

**Comment by mbaechler** [comment]
test this please

File: mpt/impl/smtp/cassandra-rabbitmq-swift/src/test/resources/logback.xml [hunk 2]
```diff
@@ -0,0 +1,30 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (aduprat) [inline mpt/impl/smtp/cassandra-rabbitmq-swift/src/test/resources/logback.xml]: Rename it to `logback-test.xml`

Comment (mbaechler) [reply mpt/impl/smtp/cassandra-rabbitmq-swift/src/test/resources/logback.xml]: I removed it, it's not loaded by logback anyway because of test-jar mess

**Pull request merged**